### PR TITLE
Add Python 3.12 to the test matrix.

### PR DESCRIPTION
**Pull request by tim-schilling**
# Checklist:

- [x] I have added the relevant tests for this change.
- [x] I have added an item to the Pending section of ``docs/changes.rst``.

### CODE REVIEW

**Comment by tim-schilling** [comment]
This is failing due to https://github.com/python/cpython/issues/107088

File: docs/changes.rst [hunk 1]
```diff
@@ -4,6 +4,8 @@ Change log
 Pending
 -------
 
+* Add Python 3.12 to test matrix.
```
Comment (matthiask) [inline docs/changes.rst:7]:
```suggestion
* Added Python 3.12 to test matrix.
```

We mostly use past tense, I think.

**Review by matthiask — APPROVED** [review]

**Comment by tim-schilling** [comment]
This is blocked by https://github.com/psycopg/psycopg/issues/626

We could bypass this by using `psycopg`

**Review by adamchainz — COMMENTED** [review]
Looks like psycopg released a compatible package so the CI can be rerun.

**Pull request merged**